File: src/main/java/com/github/dkurata38/open_web_library/domain/book/BookSummary.java
package com.github.dkurata38.open_web_library.domain.book;

import lombok.Value;

import java.time.LocalDate;

@Value
public class BookSummary {
	private final ISBN isbn;
	private final String title;
	private final String author;
	private final String publisher;
	private final LocalDate publishedDate;
	private final String coverImageUrl;
}

File: domain/src/main/kotlin/com/github/dkurata38/open_web_library/domain/book/BookSummary.kt
package com.github.dkurata38.open_web_library.domain.book

import java.time.LocalDate

data class BookSummary(
		val isbn: ISBN,
		val title: String,
		val author: String,
		val publisher: String,
		val publishedDate: LocalDate,
		val coverImageUrl: String
)
